esp/platform/espcontext.cpp

```diff
@@ -454,6 +454,15 @@ class CEspContext : public CInterface, implements IEspContext
             m_txSummary->append("total", m_processingTime, "ms");
         }
     }
+    virtual void addTraceSummaryCumulativeTime(const char* name, unsigned __int64 time)
+    {
+        if (m_txSummary)
+            m_txSummary->updateTimer(name, time);
```

Comment:
NULL and empty names will be rejected by updateTimer, which relies on getTimer to validate the input. The context doesn't validate the input because I wanted the object that imposes the requirements to have sole responsibility for validation.